src/sentry/snuba/metrics/fields/base.py

```diff
@@ -1653,10 +1641,12 @@ def generate_bottom_up_dependency_tree_for_metrics(
 
 
 def get_derived_metrics(exclude_private: bool = True) -> Mapping[str, DerivedMetricExpression]:
-    # ToDo(ahmed): Modify the behaviour here to determine whether a metric is private or not
-    #  based on whether it is added to the public naming layer or not
     return (
-        {key: value for (key, value) in DERIVED_METRICS.items() if not value.is_private}
+        {
+            mri: expression
+            for (mri, expression) in DERIVED_METRICS.items()
+            if not is_mri_private(mri)
```

Comment:
not a blocker for this PR, but for me sounds like a method of mri, rather than just a function.
You can imagine better design with:
```
for (mri, expression) in DERIVED_METRICS.items()
if not mri.is_private
```
kind of property or so.

Reply:
I 100% agree but the MRI is an enum with values of type string.